tests/vmi_multus_test.go

```diff
@@ -867,6 +868,52 @@ var _ = Describe("SRIOV", func() {
 		It("[test_id:3957]should connect to another machine with sriov interface over IPv6", func() {
 			pingThroughSriov("fc00::1/64", "fc00::2/64")
 		})
+
+		Context("VLANed SR-IOV network", func() {
+			var vlanedVMI1 *v1.VirtualMachineInstance
+			const (
+				cidrVlaned1            = "192.168.0.1/24"
+				cidrVlaned2            = "192.168.0.2/24"
+				cidrNonVlaned          = "192.168.0.3/24"
+				sriovVlanNetworkName   = "sriov-vlan"
+				sriovNoVlanNetworkName = "sriov-no-vlan"
+			)
+
+			BeforeEach(func() {
+				createNetwork(sriovVlanNetworkName, tests.NamespaceTestDefault, sriovConfVlanCRD)
+				createNetwork(sriovNoVlanNetworkName, tests.NamespaceTestDefault, sriovConfNoVlanCRD)
+				vlanedVMI1 = createSriovVMI([]string{sriovVlanNetworkName})
+				// manually configure IP/link on sriov interfaces because there is
+				// no DHCP server to serve the address to the guest
+				configInterface(vlanedVMI1, "eth1", cidrVlaned1, "\\#")
+			})
+
+			AfterEach(func() {
+				Expect(deleteNetwork(sriovVlanNetworkName, tests.NamespaceTestDefault)).To(Succeed())
+				Expect(deleteNetwork(sriovNoVlanNetworkName, tests.NamespaceTestDefault)).To(Succeed())
+				Expect(virtClient.VirtualMachineInstance(vlanedVMI1.Namespace).Delete(vlanedVMI1.Name, &v13.DeleteOptions{})).To(Succeed())
+			})
+
+			It("[test_id:3958]should be able to ping between two VMIs with the same VLAN over SRIOV network", func() {
+				vlanedVMI2 := createSriovVMI([]string{sriovVlanNetworkName})
+				configInterface(vlanedVMI2, "eth1", cidrVlaned2, "\\#")
+
+				By("pinging between vlanedVMI1 and vlanedVMI2 over vlan")
+				err = testPingFromVMI(vlanedVMI1, cidrToIP(cidrVlaned2), "\\#")
+				Expect(err).ToNot(HaveOccurred(), "VMI %s should be reachable", vlanedVMI2.Name)
+				Expect(virtClient.VirtualMachineInstance(vlanedVMI2.Namespace).Delete(vlanedVMI2.Name, &v13.DeleteOptions{})).To(Succeed())
```

Comment:
You need to do this at the After* fixture, because if the assertions fail, it will not be cleaned.